spyder_kernels/customize/namespace_manager.py

```diff
@@ -112,7 +112,10 @@ def __exit__(self, exc_type, exc_val, exc_tb):
             self.ns_globals.pop('__file__')
 
         if not self.current_namespace:
-            self.context_globals.update(self.ns_globals)
+            if self.context_locals is not None:
+                self.context_locals.update(self.ns_globals)
+            else:
+                self.context_globals.update(self.ns_globals)
```

Comment:
```suggestion
```

This `else` shouldn't be necessary because if `self.context_locals` is `None`, then we can't perform this `update`, right?

Reply:
Well if context_locals is none this means we are running in the globals namespace, and so it should be updated

Reply:
Sorry, I misread this line and thought it was the same in both branches of this `if`.